### ISSUE TITLE
Issue with validating array form fields

### ISSUE THREAD

**Issue by abdulhadi-lababidi98** [fdc-viktor-luft/morfi#30]
We have a situation where the form contain a field of type  `Array<string>`. And we want to validate that the array is not empty.
The intuitive solution is to do something like this:
```
type FormData = {
   products: string[],
}
const validators: FormValidation<FormData> = {
    products: {
        onSubmit: (value) => (value !== undefined && value.length !== 0 ? undefined : 'Validation.EntityRequired'),
    }
}
```

But this results in the following error:
![image](https://github.com/fdc-viktor-luft/morfi/assets/139132668/92236c4c-71aa-4e8e-ba50-82219f759704)

After investigation, I found that the FormValidation type does the following: 
```
export type FormValidation<V> = {
    [name in keyof V]?: V[name] extends object ? FormValidation<V[name]> & FieldValidation<V[name]> : FieldValidation<NonUndefined<V[name]>>;
};
```
> V[name] extends object ? FormValidation<V[name]> : ...

It recursively applies the FormValidation type to nested objects (V[name]). And arrays are objects. So it tried to apply the FormValidation for each key in the array prototype.

**Comment by fdc-viktor-luft** [comment]
@abdulhadi-lababidi98 Can you please see if this released version `morfi@2.1.1-beta.0` fixes your issue?

**Comment by abdulhadi-lababidi98** [comment]
Yes this worked perfectly!

**Comment by fdc-viktor-luft** [comment]
ok. Released now `morfi@2.1.1`
